perfact/zodbsync/tests/test_sync.py

```diff
@@ -2023,3 +2023,84 @@ def test_layer_update(self):
             self.run('layer-hash', layer)
             self.run('layer-update', ident)
             assert self.app.Test.title == 'Changed'
+
+    def test_keep_acl(self):
+        '''
+        Make sure deletions on top level acl_users are NOT synced into
+        Data.fs
+
+        User folders living somewhere else in the application may be
+        deleted though.
+        '''
+        acl_path = os.path.join(
+            self.repo.path,
+            '__root__',
+            'acl_users',
+        )
+        shutil.rmtree(acl_path)
+        self.run('playback', '/')
+
+        # this playback will fail horribly if acl_users is gone!
+        self.run('playback', '/')
+
+        # make sure acl_users in toplevel is still present
+        assert 'acl_users' in self.app.objectIds()
+
+        # now create dummy module with its own acl_users folder
+        self.app.manage_addFolder(id='some_module')
+        self.app.some_module.manage_addUserFolder()
+
+        self.run('record', '/')
+
+        assert 'acl_users' in self.app.some_module.objectIds()
+
+        module_acl = os.path.join(
+            self.repo.path,
+            '__root__',
+            'some_module',
+            'acl_users',
+        )
+        shutil.rmtree(module_acl)
+        self.run('playback', '/')
+        assert 'acl_users' not in self.app.some_module.objectIds()
+
+    def test_keep_acl_norecurse(self):
```

Comment:
Tipp: If you have two very similar tests that only differ in a few lines, you can write one test with two variants using `pytest.parametrize`.